### PR TITLE
Fix closeness tests in testOnComplexPlane for small inputs. Update expm1 inaccuracy regions.

### PR DESCRIPTION
**Pull request by pearu**
As in the title.

https://github.com/openxla/xla/pull/10376 fixes all inaccuracy issues in jax.numpy.expm1:
- [before this PR](https://github.com/pearu/complex_function_validation/blob/5756d0e5b303bb7b084f18f612f69ed3bf5547a2/mpmath_jax_results/data/expm1_MPMath_complex128_cpu_versus_JAX_complex64_cpu.txt) the inaccuracies and mismatch rate was 17.7%
- [after this PR](https://github.com/pearu/complex_function_validation/blob/5756d0e5b303bb7b084f18f612f69ed3bf5547a2/mpmath_jax_results_dev/data/expm1_MPMath_complex128_cpu_versus_JAX_complex64_cpu.txt) the inaccuracies rate is 0.1%

https://github.com/openxla/xla/pull/10503 fixes all inaccuracy issues in jax.numpy.log1p:
- [before this PR](https://github.com/pearu/complex_function_validation/blob/a9b875ffa4f6ee5bdc9c574545966aae6a6e5476/mpmath_jax_results/data/log1p_MPMath_complex128_cpu_versus_JAX_complex64_cpu.txt) the mismatch rate was 42.5%
- [after this PR](https://github.com/pearu/complex_function_validation/blob/a9b875ffa4f6ee5bdc9c574545966aae6a6e5476/mpmath_jax_results_dev/data/log1p_MPMath_complex128_cpu_versus_JAX_complex64_cpu.txt) the inaccuracies rate is 0%
- In addition to log1p, there exists accuracy improvements for arctanh:
  - jax.numpy.arctanh mistmatch [before](https://github.com/pearu/complex_function_validation/blob/a9b875ffa4f6ee5bdc9c574545966aae6a6e5476/mpmath_jax_results/data/arctanh_MPMath_complex128_cpu_versus_JAX_complex64_cpu.txt) is 55.5% and [after](https://github.com/pearu/complex_function_validation/blob/a9b875ffa4f6ee5bdc9c574545966aae6a6e5476/mpmath_jax_results_dev/data/arctanh_MPMath_complex128_cpu_versus_JAX_complex64_cpu.txt) 11.3 %

https://github.com/openxla/xla/pull/10525 fixes all inaccuracy issues in jax.numpy.tan:
- TBD

### CODE REVIEW

**Review by jakevdp — APPROVED** [review]

**Comment by jakevdp** [comment]
There are some failures in one of the CI jobs – looks like inf/nan mismatches.

**Comment by pearu** [comment]
> There are some failures in one of the CI jobs – looks like inf/nan mismatches.

Yes, I am in a process of fixing these. These emerged because the closeness checking is changed in this PR and the mapping of inaccuracy lists requires updates re the tests that passed false-positively before.

**Review by jakevdp — APPROVED** [review]

**Comment by jakevdp** [comment]
Tests are failing on internal CPU.

Overall, I think the issue here is that these tests are too brittle: they make detailed assertions about the dark corners of hardware-dependent implementation details. Rather than add more exceptions for particular backends / package versions / devices / etc. I think it would be better to make the tests simpler and less brittle. What do you think?

**Comment by pearu** [comment]
`xla_extension_version >= 244` needs to be updated as the current `xla_extension_version` value conflicts with the one in the main branch, hence the test failures.

**Comment by pearu** [comment]
> ... I think it would be better to make the tests simpler and less brittle.

I agree. I suggest replacing numpy with mpmath as a reference library.

File: tests/lax_test.py [hunk 1]
```diff
@@ -3365,87 +3365,119 @@ def f(x):
...
+        _functions_on_complex_plane,
+        jtu.dtypes.supported([np.complex64, np.complex128]),
+    ))
+  @jtu.skip_on_devices("tpu")
+  def testFailureOnComplexPlane(self, name, dtype):
+    self._testOnComplexPlaneWorker(name, dtype, 'failure')
+
+  def _testOnComplexPlaneWorker(self, name, dtype, kind):
+
+    mpmath = jtu.pytest_importorskip('mpmath', '1.3')
```
Comment (jakevdp) [inline tests/lax_test.py:3430]: Let's make `mpmath` a required test-time dependency to remove the need for this `importorskip` logic.

Comment (pearu) [reply tests/lax_test.py]: Done.

Comment (pearu) [reply tests/lax_test.py]: This going to be a problem for `[build with 3.9 (py 3.9 on ubuntu-20.04-16core, x64=1)](https://github.com/google/jax/actions/runs/8378976633/job/22944817602?pr=20144#logs)` that does not have mpmath installed.

Comment (pearu) [reply tests/lax_test.py]: Resolved by adding mpmath to build/test-requirements.txt

**Comment by pearu** [comment]
Created https://github.com/google/jax/pull/20373 to resolve confusion in import/copybara.

**Review by jakevdp — APPROVED** [review]

**Comment by pearu** [comment]
Closing as https://github.com/google/jax/pull/20373 has landed.

The items in the description will be addressed in follow-up PRs.

**Comment by reedwm** [comment]
@pearu, are the XLA PRs that address the inaccuracies ready for review? That is, are the following PRs ready for review?
* https://github.com/openxla/xla/pull/8589
* https://github.com/openxla/xla/pull/10376
* https://github.com/openxla/xla/pull/10503
* https://github.com/openxla/xla/pull/10525
